partiql-value/src/lib.rs

```diff
@@ -302,6 +302,8 @@ impl NullableEq for Value {
 impl NullableOrd for Value {
     type Output = Self;
 
+    // TODO: comparison is not right for data type mismatches. Equality permits mistyped arguments
```

Comment:
Will address this in a future PR. Don't want to over-expand the scope for this PR